ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm

```diff
@@ -58,4 +60,187 @@ - (void)testShouldIgnoreVnodeType {
     XCTAssertFalse(ShouldIgnoreVnodeType(static_cast<vtype>(1000), testVnode.get()));
 }
 
+- (void)testFileFlaggedInRoot {
+    bool fileFlaggedInRoot;
+    shared_ptr<mount> testMount = mount::Create();
+    shared_ptr<vnode> testVnode = vnode::Create(testMount, "/foo");
+    
+    testVnode->attrValues.va_flags = FileFlags_IsInVirtualizationRoot;
+    XCTAssertTrue(TryGetFileIsFlaggedAsInRoot(testVnode.get(), NULL, &fileFlaggedInRoot));
+    XCTAssertTrue(fileFlaggedInRoot);
+    
+    testVnode->attrValues.va_flags = FileFlags_IsEmpty;
+    XCTAssertTrue(TryGetFileIsFlaggedAsInRoot(testVnode.get(), NULL, &fileFlaggedInRoot));
+    XCTAssertFalse(fileFlaggedInRoot);
+    
+    testVnode->attrValues.va_flags = FileFlags_Invalid;
+    XCTAssertTrue(TryGetFileIsFlaggedAsInRoot(testVnode.get(), NULL, &fileFlaggedInRoot));
+    XCTAssertFalse(fileFlaggedInRoot);
+    
+    testVnode->attrValues.va_flags = 0x00000100;
+    XCTAssertTrue(TryGetFileIsFlaggedAsInRoot(testVnode.get(), NULL, &fileFlaggedInRoot));
+    XCTAssertFalse(fileFlaggedInRoot);
+
+    testVnode->errors.getattr = EBADF;
+    XCTAssertFalse(TryGetFileIsFlaggedAsInRoot(testVnode.get(), NULL, &fileFlaggedInRoot));
+}
```

Comment:
Consider also adding a test that sets `va_flags` to `FileFlags_IsInVirtualizationRoot` OR-ed (`|`) with something else and assert that `fileFlaggedInRoot` is true.

Reply:
FWIW, possible OS-defined values for flags are:

 * `UF_NODUMP`
 * `UF_IMMUTABLE`
 * `UF_APPEND`
 * `UF_OPAQUE`
 * `UF_HIDDEN`
 * `SF_ARCHIVED`
 * `SF_IMMUTABLE`
 * `SF_APPEND`

See the manpage for `chflags(2)`. These values should be defined in `<sys/stat.h>`.